src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -3,19 +3,17 @@
 
 using System.ComponentModel;
 using System.IO;
+using System.Resources.Extensions.BinaryFormat;
 using System.Runtime.Serialization;
 using System.Runtime.Serialization.Formatters.Binary;
 
 namespace System.Resources.Extensions
 {
     public partial class DeserializingResourceReader
     {
-        private bool _assumeBinaryFormatter;
+        private static readonly bool UseBinaryFormatter = AppContext.TryGetSwitch("System.Resources.Extensions.UseBinaryFormatter", out bool isEnabled) && isEnabled;
 
-// Issue https://github.com/dotnet/runtime/issues/39292 tracks finding an alternative to BinaryFormatter
-#pragma warning disable SYSLIB0011
-        private BinaryFormatter? _formatter;
-#pragma warning restore SYSLIB0011
+        private bool _assumeBinaryFormatter;
```

Comment:
I don't see this field being used?

Reply:
It's used in parts of the file that have not been modified:

https://github.com/dotnet/runtime/blob/b57bc2af4714d73ca759e9d7c5bc65c1293236b3/src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs#L31

https://github.com/dotnet/runtime/blob/b57bc2af4714d73ca759e9d7c5bc65c1293236b3/src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs#L117-L119